jasper_reports/JasperReports/jasper_report_config.py

```diff
@@ -0,0 +1,263 @@
+                            self.cr, self.uid, [self.uid], self.context)
+                    elif self.data.get('data_source', 'model') == 'records':
+                        generator = CsvRecordDataGenerator(
+                            sub_report, self.data['records'])
+                    else:
+                        generator = CsvBrowseDataGenerator(
+                            sub_report, self.model, self.env, self.cr,
+                            self.uid, self.ids, self.context)
+                    generator.generate(sub_report_data_file)
+
+            # Call the external java application that will generate the
+            # PDF file in outputFile
+            pages = self.execute_report(
+                data_file, output_file, sub_report_data_files)
+
+            elapsed = (time.time() - start) / 60
+            logger.info("ELAPSED: %f" % elapsed)
+
+            # Read data from the generated file and return it
+            with open(output_file, 'rb') as f:
+                data = f.read()
+
+            # Remove all temporary files created during the report
+            if tools.config['jasperunlink']:
+
+                for f in self.temporary_files:
+                    try:
+                        os.unlink(f)
+                    except os.error:
+                        logger.warning("Could not remove file '%s'." % f)
+
+            self.temporary_files = []
+
+            if self.context.get('return_pages'):
+                return data, self.output_format, pages
+            else:
+                return data
+
+    def path(self):
+        return os.path.abspath(os.path.dirname(__file__))
+
+    def addons_path(self, path=False):
+        if path:
+            report_module = path.split(os.path.sep)[0]
+
+            for addons_path in tools.config['addons_path'].split(','):
+                if os.path.lexists(addons_path + os.path.sep + report_module):
+                    return os.path.normpath(addons_path + os.path.sep + path)
+
+        return os.path.dirname(self.path())
+
+    def system_user_name(self):
+        if os.name == 'nt':
+            import win32api
+            return win32api.GetUserName()
+        else:
+            import pwd
+            return pwd.getpwuid(os.getuid())[0]
+
+    def dsn(self):
+        host = tools.config['db_host'] or 'localhost'
+        port = tools.config['db_port'] or '5432'
+        db_name = self.cr.dbname
+        return 'jdbc:postgresql://%s:%s/%s' % (host, port, db_name)
+
+    def user_name(self):
+        # To allow all users to get db_user we have to call it with sudo
+        user_name = self.env['ir.config_parameter'].sudo().get_param(
+            'db_user') or self.system_user_name()
+        return tools.config['db_user'] or user_name
+
+    def password(self):
+        # To allow all users to get db_password we have to call it with sudo
+        password = self.env['ir.config_parameter'].sudo().get_param(
```

Comment:
`db_user` and `db_password` can be fetched from odoo config file, Check this in odoo/tools.config .py